apps/sms/views/conversation/js/compose.js

```diff
@@ -487,25 +469,16 @@ var Compose = (function() {
       return state.empty;
     },
 
-    /**
-     * Lock composer when size limit is reached.
-     */
-    lock: function() {
-      state.locked = true;
-      dom.attachButton.disabled = true;
-    },
-
-    /**
-     * Unlock composer when size is decreased again.
-     */
-    unlock: function() {
-      state.locked = false;
-      dom.attachButton.disabled = false;
+    setupLock({ canSend, canEdit, getType }) {
```

Comment:
The main thing that bothered me was that undefined parameter actually affected lock properties... But I think it's the best and the safest option I have right now, so I'm happy that you're fine with it :)